src/dataServices/hybridDataService.ts

```diff
@@ -218,20 +218,59 @@ export class HybridDataService implements Persistence, BulkDataAccess {
         }
     }
 
+    private async deleteVersionedResource(id: string, versionId: string, bulkDataLink?: string, tenantId?: string) {
+        await this.dbPersistenceService.deleteVersionedResource(id, parseInt(versionId, 10), tenantId);
+        return {
+            id,
+            bulkDataLink,
+        };
+    }
+
     async deleteResource(request: DeleteResourceRequest) {
         this.assertValidTenancyMode(request.tenantId);
         const { resourceType, id, tenantId } = request;
-        const itemServiceResponse = await this.dbPersistenceService.readResource({ resourceType, id, tenantId });
-        const { versionId } = itemServiceResponse.resource.meta;
-        const { bulkDataLink } = itemServiceResponse.resource;
-        if (bulkDataLink) {
-            const [, deleteResponse] = await Promise.all([
-                S3ObjectStorageService.deleteObject(bulkDataLink),
-                this.dbPersistenceService.deleteVersionedResource(id, parseInt(versionId, 10), tenantId),
-            ]);
-            return deleteResponse;
+
+        const projectionExpression = 'id, meta, bulkDataLink';
+        const itemServiceResponses = await this.dbPersistenceService.readAllResourceVersions(
+            {
+                resourceType,
+                id,
+                tenantId,
+            },
+            projectionExpression,
+        );
+
+        const deleteDbItemPromsises: Array<Promise<any>> = [];
+        itemServiceResponses.forEach((element: any) => {
+            const { versionId } = element.meta;
+            const { bulkDataLink } = element;
+            deleteDbItemPromsises.push(this.deleteVersionedResource(id, versionId, bulkDataLink, tenantId));
+        });
+
+        const deletionDbItemResults = await Promise.all(deleteDbItemPromsises.map((p) => p.catch((e) => e)));
```

Comment:
I found the following solution:
```js
const successes = await Promise.allSettled(resDelPromises)
    .filter(p => p.status === 'fulfilled')
    .map(p => p.value);
const bulkDelPromises = successes
    .filter(res => res.bulkDataLink)
    .map(res => S3ObjectStorageService.deleteObject(res.bulkDataLink));
await Promise.all(bulkDelPromises);
```

Reply:
I am not sure in which order `await` will be applied. May be one should do like this:
```js
    const resDelSuccesses = Promise.allSettled(resDelPromises).wait()
    ...
```

Reply:
I've seen allSetteled, the other solution I know that is working. That's why I'd keep the existing Promise.all.
But the filter().map() I'll add